protocol/src/lib.rs

```diff
@@ -206,14 +209,32 @@ impl FromStr for ModelDescription {
     fn from_str(s: &str) -> std::result::Result<Self, Self::Err> {
         let split = s.split("/");
         let tokens: Vec<String> = split.map(|v| v.to_string()).collect();
-        if tokens.len() < 3 || tokens.iter().any(|v| v.is_empty()) {
+        if tokens.len() < 3 || tokens[..2].iter().any(|v| v.is_empty()) {
             return Err(());
         }
+        let maybe_endpoint = if tokens.len() >= 3 {
+            let tail = tokens[3..].join("/");
+            // TODO: Maybe use the url crate for query string parsing? (see: https://docs.rs/url/1.4.0/url/form_urlencoded/fn.parse.html)
```

Comment:
so you're thinking we would use `form_urlencoded::parse` so we can handle additional parameters, if need be? it's a good idea. happy to keep this as a todo for now :+1:

Reply:
I missed this one - Yeah something like that would work if we keep the format the same (see my comment here https://github.com/DJAndries/llmvm/pull/2#issuecomment-1801789056)